onnx/onnx.in.proto

```diff
@@ -389,6 +390,15 @@ message ModelProto {
   //
   // If this field is empty, the training behavior of the model is undefined.
   repeated TrainingInfoProto training_info = 20;
+
+  // A list of function protos local to the model.
+  //
+  // The opset imports in FunctionProto should not conflict with the ones in 
+  // model proto and should be mergeable. Example, if the function proto imports domain  
+  // 'A' opset 10 whereas model proto also imports domain 'A' but with a different opset
+  // then this is a merge conflict.
+  repeated FunctionProto functions = 25;
```

Comment:
How does the model refer to these functions? If they were in an operator set, presumably there would be a domain, but I don't see a domain here.

Should this field be an OperatorSetProto (or some subset of that proto)?

Reply:
FunctionProto has a "name" field. The model compares this with the op_type field from NodeProto.

Rama and I were discussing whether we want domain qualified names, meaning should we add domain field to the functionproto as well... we can do that if we think just the name field wont be enough...

Reply:
Two possible options: 
(a) We add a domain field to FunctionProto. (We then need to decide whether we want to allow these in-model functions to override out-of-model functions defined in opschema registries.)
(b) Reserve a domain for in-model functions (like 'onnx.ai.local') if we want to avoid the above.

Reply:
added domain field for FunctionProto